Assets/Scripts/Character/heat/Warmth.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class Warmth : CharacterBehaviour
+{
+    public const int MaxWarmth = 100;
+
+    public float damageTakenDelaySeconds;
+    public int damagePercentage;
+
+
+    public UnityEvent deathEvent;
+
+    [HideInInspector] public float heat;
+
+    private static readonly int DyingHash = Animator.StringToHash("dying");
+
+    private float slowDamageMultiplier;
+
+    protected override void Awake()
+    {
+        base.Awake();
+        slowDamageMultiplier = 1;
+        heat = MaxWarmth;
+        StartCoroutine(ChillDown());
+    }
+
+    private void TakeDamage()
+    {
+        heat -= damagePercentage * slowDamageMultiplier;
+        CheckDeath();
+    }
+
+    private void CheckDeath()
+    {
+        if (heat <= 0)
+        {
+            Animator.SetBool(DyingHash, true);
+            deathEvent.Invoke();
+            Debug.Log("no chill!!\n [character is dead]");
+        }
+    }
+
+    public void SlowChill(bool startOrEnd, float slowDamageMultiplier_)
```

Comment:
I don't like booleans that completely change the method. Split up to two methods, one for start and one for end